http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -444,6 +461,22 @@ protected AuthChallenge resolveCode(String code) {
             TokenValidator.VerifiedTokens verifiedTokens = tokenValidator.parseAndVerifyToken(idTokenString, tokenString);
             idToken = verifiedTokens.getIdToken();
             token = verifiedTokens.getAccessToken();
+
+            String stateCookieValue = getCookieValue(deployment.getStateCookieName());
+            String[] cookieArr = stateCookieValue.split(";");
+            String sessionIdValue = "";
+            for(int i=0; i < cookieArr.length; i++) {
+                String cookie = cookieArr[i].trim();
+                if (cookie.startsWith(SESSION_ID+"=")) {
```

Comment:
Just to check, is this needed? I don't see a similar check being done in the code that obtains the state cookie value,